linkd-lib/Cargo.toml

```diff
@@ -0,0 +1,42 @@
+[package]
+name    = "linkd-lib"
+version = "0.1.0"
+edition = "2018"
+license = "GPL-3.0-or-later"
+authors = [
+  "xla <self@xla.is>",
+]
+
+[lib]
+doctest = true
+test    = false
+
+[dependencies]
+anyhow              = "1.0"
+base64              = "0.13"
+env_logger          = "0.9"
+futures             = "0.3"
+lazy_static         = "1.4"
+log                 = "0.4"
+nix                 = "0.22"
+signal-hook         = { version = "0.3", default-features = false }
+signal-hook-tokio   = { version = "0.3", features = [ "futures-v0_3" ] }
```

Comment:
This does not support Windows, as it uses a "pipe" which actually is a tokio `UnixStream` for signal delivery. It saves a thread, yes, but I'm not sure this is overall cheaper.

Reply:
Was mostly optimising for ergonomics as this fits better with the style of spawning a task per subroutine. While the failing windows tests haunted me in my dreams the last days I concluded it might be worthwhile porting the parked thread with atomics approach we have in other places over and give the impression that this project works on Windows after all. Maybe then @FintanH stops calling me out on it.